src/crimson/net/ProtocolV2.cc

```diff
@@ -944,6 +944,11 @@ void ProtocolV2::execute_connecting()
             return client_reconnect();
           }
         }).then([this] (next_step_t next) {
+          if (unlikely(state != state_t::CONNECTING)) {
```

Comment:
can we use `assert()` instead? if this should never happen if the code works correctly, and we cannot recover from it. i'd prefer we don't have this in release build at all.

Reply:
No, this is not for debugging. It can happen because close and replace are preemptive. Like user close the connection while it is still in the middle of `state_t::CONNECTING`, or another connection replaces this one during its `state_t::CONNECTING`. In these cases,`execute_connecting()` is obsolete and should be aborted.